coordinator/src/node.rs

```diff
@@ -213,7 +210,29 @@ impl Node {
                     "Opening positions is disabled"
                 );
 
-                self.open_position(trade_params, fee_payment_hash).await?;
+                let channel_details = self.get_counterparty_channel(trade_params.pubkey)?;
+
+                let user_channel_id = Uuid::from_u128(channel_details.user_channel_id).to_string();
+                let connection = &mut self.pool.get()?;
+                let channel =
+                    db::channels::get(&user_channel_id, connection)?.with_context(|| {
+                        format!(
+                            "Couldnt find shadow channel. trader_id={}, user_channel_id={}",
+                            trade_params.pubkey, channel_details.user_channel_id
+                        )
+                    })?;
+
+                let liquidity_option =
+                    db::liquidity_options::get(connection, channel.liquidity_option_id)?;
+                let coordinator_leverage = liquidity_option.coordinator_leverage;
```

Comment:
Alternatively you could do a `join` on db level and spare one db request.

Reply:
I am opting to skip that suggestion 🙈